tests/util/kube_utils.go

```diff
@@ -641,17 +641,30 @@ func CheckDeployments(namespace string, timeout time.Duration, kubeconfig string
 func FetchAndSaveClusterLogs(namespace string, tempDir string, kubeconfig string) error {
 	var multiErr error
 	fetchAndWrite := func(pod string) error {
+		// Log the description; if we fail to get the logs it may help
+		describeCmd := fmt.Sprintf("kubectl -n %s describe pod %s --kubeconfig=%s",
+			namespace, pod, kubeconfig)
+		describeOutput, errDescribe := Shell(describeCmd)
+		if errDescribe != nil {
+			log.Warnf("Error getting description for pod %s: %v\n", pod, errDescribe)
+			// don't bail, keep going
+		} else {
+			log.Info(describeOutput)
+		}
+
 		cmd := fmt.Sprintf(
 			"kubectl get pods -n %s %s -o jsonpath={.spec.containers[*].name} --kubeconfig=%s", namespace, pod, kubeconfig)
 		containersString, err := Shell(cmd)
 		if err != nil {
+			log.Warnf("Error getting containers for pod %s: %v\n", pod, err)
```

Comment:
Are we not logging these elsewhere?

Reply:
The current logging is underwhelming.  It merely says "Log saving incomplete" and then something like "* command failed: "Error from server (BadRequest): container \"prometheus\" in pod \"prometheus-7b9868c46b-qxn42\" is waiting to start: "

My preference is to log the problem close to the error.  If you don't want the additional log line would it be reasonable to prefix the error returned?  Too often the error from `Shell()` is just "Error from server (BadRequest)".  The general nature of the log messages slows down debugging.